clients/src/main/java/org/apache/kafka/common/config/ConfigException.java

```diff
@@ -33,6 +33,10 @@ public ConfigException(String name, Object value) {
         this(name, value, null);
     }
 
+    public ConfigException(String message, Throwable cause) {
+        super(message, cause);
```

Comment:
It might make sense to redirect the call to the other constructor `ConfigException(String name, Object value, String message)` with `name, null, null` here if `cause` is null, because we know that any caller passing a real `Throwable` won't pass a null.

Reply:
Thanks for the suggestion. We're very restricted with what we can do within a constructor unfortunately as call to `super()` must be the first statement if it's to be used. This is due to change with Java 22 with [JEP 447](https://openjdk.org/jeps/447) landing but I doubt we can add a conditional at the moment.

Reply:
That is a good point! Thanks for the link to the JEP, quite interesting!